sources/packages/backend/apps/api/src/services/application-restriction-bypass/application-restriction-bypass.service.ts

```diff
@@ -49,4 +82,380 @@ export class ApplicationRestrictionBypassService extends RecordDataModelService<
       order: { createdAt: "DESC" },
     });
   }
+
+  /**
+   * Gets an application restriction bypass by id.
+   * @param id id of the application restriction bypass to retrieve.
+   * @returns application restriction bypass.
+   */
+  getApplicationRestrictionBypass(
+    id: number,
+  ): Promise<ApplicationRestrictionBypass> {
+    return this.repo.findOne({
+      select: {
+        id: true,
+        isActive: true,
+        studentRestriction: {
+          id: true,
+          restriction: {
+            restrictionCode: true,
+          },
+        },
+        creationNote: true as unknown,
+        removalNote: true as unknown,
+        bypassCreatedBy: true as unknown,
+        bypassCreatedDate: true,
+        bypassRemovedBy: true as unknown,
+        bypassRemovedDate: true,
+        bypassBehavior: true,
+      },
+      relations: {
+        studentRestriction: { restriction: true },
+        creationNote: true,
+        removalNote: true,
+        bypassCreatedBy: true,
+        bypassRemovedBy: true,
+      },
+      where: {
+        id,
+        application: {
+          applicationStatus: Not(
+            In([
+              ApplicationStatus.Draft,
+              ApplicationStatus.Cancelled,
+              ApplicationStatus.Overwritten,
+            ]),
+          ),
```

Comment:
Why would be require some extra where condition than the ID itself?

Reply:
Can you please also remove the below condition
```ts
applicationStatus: Not(ApplicationStatus.Draft)
```
from the method `getApplicationRestrictionBypasses`. It can create confusion and it is not right.